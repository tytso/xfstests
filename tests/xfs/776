#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2022 Oracle.  All Rights Reserved.
#
# FS QA Test No. 776
#
# Populate a XFS filesystem and fuzz every btree-format attr inode field.
# Try online repair and, if necessary, offline repair,
# to test the most likely usage pattern.

. ./common/preamble
_begin_fstest dangerous_fuzzers fuzzers_bothrepair

_register_cleanup "_cleanup" BUS

# Import common functions.
. ./common/filter
. ./common/populate
. ./common/fuzzy

_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Find btree-format attr inode"
_scratch_mount
inum=$(stat -c '%i' $SCRATCH_MNT/ATTR.FMT_BTREE)
_scratch_unmount

echo "Fuzz inode"
_scratch_xfs_fuzz_metadata '' 'both'  "inode ${inum}" >> $seqres.full
echo "Done fuzzing inode"

# success, all done
status=0
exit
