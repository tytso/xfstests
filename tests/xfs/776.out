QA output created by 776
Format and mount
Misconfigure the root directory
Detect misconfigured directory
Repair misconfigured directory
